hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -339,7 +344,7 @@ int MDMParser::sendFormattedWithArgs(const char* format, va_list args) {
     return n;
 }
 
-bool MDMParser::_atOk(void)
+int MDMParser::_checkAtResponse(void)
```

Comment:
There are still a number of places in the code where a `_atOk()` check is used instead of a `_checkModem()`, and currently `_atOk()` does not increment `_error` by itself.  We can't add `_error++;` to the `_atOk()` timeout case or it would double increment when `_checkModem()` is called, so should we convert all `_atOk()` uses to `_checkModem()`?

Reply:
For now we are only handling any functionality that can be triggered by the system or the application (sockets, cellular signal, band select etc). The initialization logic for instance handles timeouts and failures on its own. If we are to do this type of refactoring it's a much larger task with more area of impact that will require significantly more time and not really fit for 2.0.0-rc.2.

We should queue this up for sometime in the future though.

Reply:
Isn't this just changing 4-5 `_atOk()` statements to `_checkModem()` statements?  If the AT/OK check fails, an error will be accumulated.  Without this, we are fixing most of the cases where we previously decided AT/OK checks ahead of time were important, but leaving a handful of them that can fail without bubbling up to the system layer.  For all of these, we jump to failure, so accumulating an error in those cases makes a lot of sense.

Is the larger task just the testing effort?

Reply:
Let's postpone this work until a later release. There shouldn't be any negative consequences leaving things as-is.